aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -352,11 +356,34 @@ Tensor sparse_compressed_to_dense(
     const Tensor& self,
     c10::optional<ScalarType> dtype) {
   TORCH_CHECK(
-      !dtype.has_value(), "dtype argument is not supported by sparse_csr_to_dense");
+      !dtype.has_value(),
+      "dtype argument is not supported by sparse_csr_to_dense");
   if (self.layout() == kSparseCsr) {
     Tensor dst = at::zeros(self.sizes(), self.options().layout(kStrided));
     return dst.add_(self);
   }
+  if (self.layout() == kSparseBsr) {
+    TORCH_CHECK(self.dim() == 2, "Can only convert 2D SparseBsr to Strided.");
+    Tensor indices = at::_convert_indices_from_csr_to_coo(
+        self.crow_indices(), self.col_indices(), false, false);
+    auto values = self.values();
+    int64_t blocksize[2] = {values.size(-2), values.size(-1)};
+    IntArrayRef expanded_size(
```

Comment:
The logical and simple fix to the issues we've seen should be resolved by using DimVector instead of IntArrayRef... - IntArrayRef immediately loses access to the values it references in the way this is defined. To some extent it's debatable whether IntArrayRef should allow this kind of constructor to begin with.